src/main/java/com/easypost/easyvcr/Censors.java

```diff
@@ -204,4 +212,119 @@ public String applyQueryParametersCensors(String url) {
+            return dictionary;
+        }
+
+        Map<String, Object> censoredBodyDictionary = new HashMap<>();
+
+        for (Map.Entry<String, Object> entry : dictionary.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (keyShouldBeCensored(key, this.bodyParamsToCensor)) {
+                if (value == null) {
+                    // don't need to worry about censoring something that's null
+                    // (don't replace null with the censor string)
+                    continue;
+                } else if (Utilities.isDictionary(value)) {
+                    // replace with empty dictionary
+                    censoredBodyDictionary.put(key, new HashMap<>());
+                } else if (Utilities.isList(value)) {
+                    // replace with empty array
+                    censoredBodyDictionary.put(key, new ArrayList<>());
+                } else {
+                    // replace with censor text
+                    censoredBodyDictionary.put(key, this.censorText);
+                }
+            } else {
+                if (Utilities.isDictionary(value)) {
+                    // recursively censor inner dictionaries
+                    try {
+                        // change the value if can be parsed as a dictionary
+                        value = applyBodyCensors((Map<String, Object>) value);
+                    } catch (ClassCastException e) {
+                        // otherwise, skip censoring
+                    }
+                } else if (Utilities.isList(value)) {
+                    // recursively censor list elements
+                    try {
+                        // change the value if can be parsed as a list
+                        value = applyBodyCensors((List<Object>) value);
+                    } catch (ClassCastException e) {
+                        // otherwise, skip censoring
+                    }
+                }
+
+                censoredBodyDictionary.put(key, value);
+            }
+        }
+
+        return censoredBodyDictionary;
+    }
+
+    private String censorJsonBodyParameters(String body) {
+        Map<String, Object> bodyDictionary;
+        try {
+            bodyDictionary = Serialization.convertJsonToObject(body, Map.class);
+            Map<String, Object> censoredBodyDictionary = applyBodyCensors(bodyDictionary);
+            return censoredBodyDictionary == null ? body : Serialization.convertObjectToJson(censoredBodyDictionary);
+        } catch (Exception ex) {
+            // body is not a JSON dictionary
+            try {
+                List<Object> bodyList = Serialization.convertJsonToObject(body, List.class);
+                List<Object> censoredBodyList = applyBodyCensors(bodyList);
+                return censoredBodyList == null ? body : Serialization.convertObjectToJson(censoredBodyList);
+            } catch (Exception ex2) {
```

Comment:
Why `ex2`?

Shouldn't we also raise an error instead of failing silently by returning the body. If the user asked this function to censor something and they trust it'll censor, when they get back a body non-censored and trusted, they could happily commit sensitive info because we didn't warn that this function and their data aren't compatible.

Reply:
Sounds good. EasyVCR Java will only apply censors to JSON data for the time being. Attempting to apply censors to non-JSON data will throw an exception.